test/org/sagebionetworks/bridge/play/controllers/BaseControllerTest.java

```diff
@@ -138,7 +131,60 @@ public void doesNotThrowErrorWhenUserAgentStringInvalid() throws Exception {
         assertNull(info.getSdkName());
         assertNull(info.getSdkVersion());
     }
-    
+
+    @Test
+    public void setWarningHeaderWhenNoUserAgent() throws Exception {
+        mockPlayContext();
+
+        ClientInfo info = new SchedulePlanController().getClientInfoFromUserAgentHeader();
+
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_USER_AGENT);
+
+        assertNull(info.getAppName());
+        assertNull(info.getAppVersion());
+        assertNull(info.getOsName());
+        assertNull(info.getOsVersion());
+        assertNull(info.getSdkName());
+        assertNull(info.getSdkVersion());
+    }
+
+    @Test
+    public void setWarningHeaderWhenEmptyUserAgent() throws Exception {
+        mockPlayContext();
+        mockHeader(USER_AGENT, "");
+
+        ClientInfo info = new SchedulePlanController().getClientInfoFromUserAgentHeader();
+
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_USER_AGENT);
+
+        assertNull(info.getAppName());
+        assertNull(info.getAppVersion());
+        assertNull(info.getOsName());
+        assertNull(info.getOsVersion());
+        assertNull(info.getSdkName());
+        assertNull(info.getSdkVersion());
+    }
+
+    @Test
+    public void setWarningHeaderWhenNullUserAgent() throws Exception {
+        mockPlayContext();
+        mockHeader(USER_AGENT, null);
+
+        ClientInfo info = new SchedulePlanController().getClientInfoFromUserAgentHeader();
+
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_USER_AGENT);
+
+        assertNull(info.getAppName());
+        assertNull(info.getAppVersion());
+        assertNull(info.getOsName());
+        assertNull(info.getOsVersion());
+        assertNull(info.getSdkName());
+        assertNull(info.getSdkVersion());
+    }
```

Comment:
Add a verification that we don't set the header if the User-Agent string parses correctly